workers/unity/Packages/com.improbable.gdk.core/Worker/ViewDiff.cs

```diff
@@ -0,0 +1,307 @@
+                    }
+
+                    if (typeof(ICommandDiffStorage).IsAssignableFrom(type) && !type.IsAbstract)
+                    {
+                        var instance = (ICommandDiffStorage) Activator.CreateInstance(type);
+
+                        commandStorageList.Add(instance);
+                        if (!componentIdToCommandIdToStorage.TryGetValue(instance.GetComponentId(),
+                            out var commandIdToStorage))
+                        {
+                            commandIdToStorage = new Dictionary<uint, ICommandDiffStorage>();
+                            componentIdToCommandIdToStorage.Add(instance.GetComponentId(), commandIdToStorage);
+                        }
+
+                        commandIdToStorage.Add(instance.GetCommandId(), instance);
+                    }
+                }
+            }
+        }
+
+        public void Clean()
+        {
+            foreach (var storage in componentStorageList)
+            {
+                storage.Clear();
+            }
+
+            foreach (var storage in commandStorageList)
+            {
+                storage.Clear();
+            }
+
+            worldCommandStorage.Clear();
+            entitiesAdded.Clear();
+            entitiesRemoved.Clear();
+            logs.Clear();
+            metrics.Clear();
+            flags.Clear();
+        }
+
+        public void AddEntity(long entityId)
+        {
+            if (!entitiesRemoved.Remove(new EntityId(entityId)))
+            {
+                entitiesAdded.Add(new EntityId(entityId));
+            }
+        }
+
+        public void RemoveEntity(long entityId)
+        {
+            if (!entitiesAdded.Remove(new EntityId(entityId)))
+            {
+                entitiesRemoved.Add(new EntityId(entityId));
+            }
+        }
+
+        public void AddComponent<T>(T component, long entityId, uint componentId) where T : ISpatialComponentUpdate
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var storage))
+            {
+                throw new ArgumentException("Unknown component ID");
+            }
+
+            ((IDiffComponentAddedStorage<T>) storage).AddEntityComponent(entityId, component);
+        }
+
+        public void RemoveComponent(long entityId, uint componentId)
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var storage))
+            {
+                throw new ArgumentException("Unknown component ID");
+            }
+
+            storage.RemoveEntityComponent(entityId);
+        }
+
+        public void SetAuthority(long entityId, uint componentId, Authority authority)
+        {
+            if (!componentIdToComponentStorage.TryGetValue(componentId, out var authorityStorage))
+            {
+                throw new ArgumentException("Unknown component");
+            }
+
+            ((IDiffAuthorityStorage) authorityStorage).AddAuthorityChange(
```

Comment:
Just wondering, why does this need a cast?

Reply:
This one strictly can be rolled into the parent one as it doesn't need new type information. I just prefer to keep the sections isolated. I suspect there are some inconsistencies though.